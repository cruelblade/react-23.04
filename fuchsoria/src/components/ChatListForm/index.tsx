import React, { useState, FormEvent } from 'react';
import { Input, Button } from 'antd';
import styles from './styles.module.scss';

export default function ChatListForm({ handleCreate }: { handleCreate: (title: string) => void }) {
  const [title, setTitle] = useState('');

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault();

    handleCreate(title);
  };

  return (
    <form className={styles.form} onSubmit={handleSubmit}>
      <Input type="text" placeholder="Введите название" onChange={(event) => setTitle(event.target.value)} />
      <Button htmlType="submit">Создать чат</Button>
    </form>
  );
}
