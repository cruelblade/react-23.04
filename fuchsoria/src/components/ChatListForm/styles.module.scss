.form {
  border-top: 1px #d6d6d6 solid;
  padding: 10px;
  display: flex;
}