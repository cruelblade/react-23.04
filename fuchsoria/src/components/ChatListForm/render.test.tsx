import React from 'react';
import { render } from '@testing-library/react';
import ChatListForm from '../ChatListForm';

describe('<ChatListForm />', () => {
  it('Component should render', () => {
    const { container } = render(<ChatListForm handleCreate={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
