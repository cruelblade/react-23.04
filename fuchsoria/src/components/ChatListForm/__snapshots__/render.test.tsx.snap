// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatListForm /> Component should render 1`] = `
<div>
  <form
    class="form"
  >
    <input
      class="ant-input"
      placeholder="Введите название"
      type="text"
      value=""
    />
    <button
      class="ant-btn"
      type="submit"
    >
      <span>
        Создать чат
      </span>
    </button>
  </form>
</div>
`;
